hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -1281,13 +1281,30 @@ public void removeRecord0(Data key, @Nonnull Record record, boolean backup) {
         storage.removeRecord(key, record);
     }
 
+    public Record removeByKey(Data key, boolean backup) {
+        Record record = getRecord(key);
+        mutationObserver.onRemoveRecord(key, record, backup);
+        removeKeyFromExpirySystem(key);
+        storage.removeRecord(key, record);
+        return record;
+    }
+
     @Override
     public Record getRecordOrNull(Data key, boolean backup) {
         long now = getNow();
         return getRecordOrNull(key, now, backup);
     }
 
     public Record getRecordOrNull(Data key, long now, boolean backup) {
+        return getRecordOrNull(key, now, backup, false);
+    }
+
+    public Record getRecordOrNull(Data key, boolean backup, boolean noCaching) {
```

Comment:
could you please javaDoc what is `noCaching` and why is it useful?

Reply:
Added in https://github.com/hazelcast/hazelcast/pull/26065/commits/aa0233867b6cbef099b83ffc486b7dc17f495390